src/batched_inference.py

```diff
@@ -156,16 +157,143 @@ def infer(self):
         )
         generation_output = self._model.generate(
             input_ids=input_tokens,
-            max_new_tokens=self._batch_size
+            max_new_tokens=self._batch_size,
+            do_sample=True
```

Comment:
Why is the base class modified?  Also seems like this parameter is not being used.

Reply:
`max_new_tokens` was a parameter already in `batched_inference.py` from your implementation. You were missing a comma after `max_new_tokens=self._batch_size`, and hence I added it.